packages/gatsby/lib/query-runner/file-parser.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+const fs = require(`fs`)
+const crypto = require(`crypto`)
+
+// Traverse is a es6 module...
+import traverse from 'babel-traverse'
+import * as types from 'babel-types'
+const babylon = require(`babylon`)
+const Bluebird = require(`bluebird`)
+
+const apiRunnerNode = require(`../utils/api-runner-node`)
+const { getGraphQLTag } = require(`../utils/babel-plugin-extract-graphql`)
+
+import type { DocumentNode, DefinitionNode } from 'graphql'
+
+const readFileAsync = Bluebird.promisify(fs.readFile)
+
+function getAssignedIdenifier(path) {
+  let property = path.parentPath
+  while (property) {
+    if (types.isVariableDeclarator(property)) return property.node.id.name
+    property = property.parentPath
+  }
+}
+
+async function parseToAst(filePath, fileStr) {
+  let ast
+
+  // Preprocess and attempt to parse source; return an AST if we can, log an
+  // error if we can't.
+  const transpiled = await apiRunnerNode(`preprocessSource`, {
+    filename: filePath,
+    contents: fileStr,
+  })
+
+  if (transpiled.length) {
+    for (const item of transpiled) {
+      try {
+        const tmp = babylon.parse(item, {
+          sourceType: `module`,
+          plugins: [`*`],
```

Comment:
side note that internally RelayCOmpiler switched from `*` to some explicit list...i'm not sure why